pkgs/build-support/rust/default.nix

```diff
@@ -204,11 +205,12 @@ stdenv.mkDerivation (args // {
 
   checkPhase = args.checkPhase or (let
     argstr = "${stdenv.lib.optionalString (checkType == "release") "--release"} --target ${rustTarget} --frozen";
+    threads = if singleTestThread == false then "$NIX_BUILD_CORES" else "1";
```

Comment:
not super relevant, but this can also be written as `if singleTestThread then "1" else "$NIX_BUILD_CORES"`